src/run_program.rs

```diff
@@ -141,18 +146,20 @@ impl<'a, D: Dialect> RunProgramContext<'a, D> {
         Ok(())
     }
 
-    fn new(allocator: &'a mut Allocator, dialect: &'a D, pre_eval: Option<PreEval>) -> Self {
+    fn new(allocator: &'a mut Allocator, dialect: &'a D, _pre_eval: Option<PreEval>) -> Self {
```

Comment:
You might want to simply do an `assert!(pre_eval.is_none());` in the case the feature isn't enabled. Or maybe even a panic that says the `pre_eval` feature isn't enabled. Or maybe `new_with_pre_eval` that only exists if the feature is enabled.

Reply:
good point. then it won't be unused as well

Reply:
seconded, but i'd say make this entry point conditional on pre_eval as a feature and make a new one present in both that doesn't accept pre_eval.
that way code that expects this to work won't fail silently.

Reply:
I made this change. please see the second commit